File: server/route/admin/knowledge.py
from fastapi import APIRouter, Query, HTTPException, status, Depends
from typing import Optional
from sqlalchemy.ext.asyncio import AsyncSession
from service.admin.knowledge import KnowledgeService
from store.database import GetDB
from schema import (
    SearchSchema,
    ResponseSchema,
    KnowledgeCreateSchema,
    KnowledgeUpdateSchema,
)

router = APIRouter(prefix="/knowledge")


@router.get("/search")
async def search_knowledge(params: SearchSchema, db: AsyncSession = GetDB):
    """获取知识点列表"""
    res = KnowledgeService.search(db, params)
    return ResponseSchema(data=res)


@router.post("/")
async def create_knowledge(knowledge: KnowledgeCreateSchema, db: AsyncSession = GetDB):
    """创建知识点"""
    id = await KnowledgeService.create(db, knowledge)
    return ResponseSchema(data=id)


@router.get("/{knowledge_id}")
async def get_knowledge(knowledge_id: int, db: AsyncSession = GetDB):
    """获取单个知识点"""
    knowledge = await KnowledgeService.get_by_id(db, knowledge_id)
    return ResponseSchema(data=knowledge)


@router.patch("/{knowledge_id}")
async def update_knowledge(
    knowledge_id: str, knowledge: KnowledgeUpdateSchema, db: AsyncSession = GetDB
):
    """更新知识点"""
    await KnowledgeService.update(db, knowledge_id, knowledge)
    return ResponseSchema()


@router.delete("/{knowledge_id}")
async def delete_knowledge(knowledge_id: str, db: AsyncSession = GetDB):
    """删除知识点"""
    await KnowledgeService.delete(db=db, knowledge_id=knowledge_id)
    return ResponseSchema()

File: admin/src/services/knowledge.ts
import { request } from '@umijs/max';

export const KnowledgeApi = {
  create: async (data: KnowledgeForm) => {
    const res = await request<ApiData<Knowledge>>('/knowledge', {
      method: 'POST',
      data,
    });
    return res.data;
  },

  update: async (id: number, data: KnowledgeUpdateForm) => {
    const res = await request<ApiData<Knowledge>>(`/knowledge/${id}`, {
      method: 'PATCH',
      data,
    });
    return res.data;
  },

  delete: async (id: number) => {
    await request<ApiData<void>>(`/knowledge/${id}`, {
      method: 'DELETE',
    });
  },

  search: async (params: KnowledgeSearchParams) => {
    const res = await request<ListApiData<Knowledge>>('/knowledge/search', { params });
    return res.data;
  },

  getQuestions: async (id: number, params?: { page?: number; size?: number }) => {
    const res = await request<ListApiData<Question>>(`/knowledge/${id}/questions`, { params });
    return res.data;
  },
};
